lib/puppet/provider/package/cisco.rb

```diff
@@ -53,7 +66,8 @@
   # this method in package.rb determines how to set ensure
   # @property_hash is empty at this point
   def properties
-    if in_guestshell? && target_host?
+    if (in_nexus? && in_guestshell? && target_host?) ||
+       (in_ios_xr? && target_host?)
```

Comment:
target_host? is used to indicate that guestshell should install to the host (native NX) instead of within the guestshell LXC. It doesn't really apply to XR as a concept at present. So this could probably just be:

```ruby
if in_ios_xr? || (in_guestshell && target_host?)
```